sys/kern/signal.c

```diff
@@ -201,6 +204,88 @@ int do_sigsuspend(proc_t *p, const sigset_t *mask) {
   return ERESTARTNOHAND;
 }
 
+int do_sigtimedwait(proc_t *p, sigset_t waitset, ksiginfo_t *kinfo,
+                    timespec_t *tsp) {
+  int timevalid = 0, error = 0, timeout = 0;
+  sigset_t saved_mask;
+  signo_t sig;
+  thread_t *td = p->p_thread;
+
+  if (tsp != NULL) {
+    if (tsp->tv_nsec < 0 || tsp->tv_nsec >= 1000000000) {
+      return EINVAL;
+    }
+    /* This is the same behaviour as in OpenBSD. */
+    if (tsp->tv_sec < 0) {
+      return EAGAIN;
+    }
```

Comment:
Could you give some resource to that behavior in OpenBSD? I can't find it.

Reply:
https://mimiker.ii.uni.wroc.pl/source/xref/NetBSD/sys/kern/sys_sig.c?r=53d22107#752

Reply:
I'm sorry to bring confusion. I've seen this behaviour in NetBSD, not OpenBSD.